storage/remote/queue_manager.go

```diff
@@ -428,11 +476,11 @@ func (s *shards) stop(deadline time.Duration) {
 	return
 }
 
-func (s *shards) enqueue(sample *model.Sample) bool {
+func (s *shards) enqueue(sample tsdb.RefSample) bool {
 	s.qm.samplesIn.incr(1)
 
-	fp := sample.Metric.FastFingerprint()
-	shard := uint64(fp) % uint64(len(s.queues))
+	// fp := sample.Metric.FastFingerprint()
+	shard := uint64(sample.Ref) % uint64(len(s.queues))
 
 	select {
 	case s.queues[shard] <- sample:
```

Comment:
Need to remove the `select` and `default` - this should block if the queue is full.